openpype/pipeline/publish/lib.py

```diff
@@ -594,6 +596,9 @@ def get_errored_instances_from_context(context):
             # When instance is None we are on the "context" result
             continue
 
+        if plugin is not None and result.get("plugin") != plugin:
+            continue
```

Comment:
It could be that this should be:
```suggestion
        if plugin is not None and isinstance(result.get("plugin"), plugin):
            continue
```
If the result["plugin"] is an instance and `plugin` is the class. I haven't tested the code itself. If it fails, then try this change ;) @mustafa-zarkash

Reply:
Why not to use `pyblish.api.instances_by_plugin(errored_instances, plugin)`?

Reply:
Because that returns the related instances - but not if those errored. We need to find only those instances that errored for this particular plug-in.

Reply:
I meant at the end of function (where are already filtered out).
```
    instances = list()
    for result in context.data["results"]:
        if result["instance"] is None:
            # When instance is None we are on the "context" result
            continue

        if result["error"]:
            instances.append(result["instance"])
    
    if plugin is None:
        return instances
    return pyblish.api.instances_by_plugin(instances, plugin)
```

Reply:
@iLLiCiTiT  
I will use `SelectInvalidAction` 
This roughly the base code to get_invalid 
```python
errored_instances = get_errored_instances_from_context(context)
instances = pyblish.api.instances_by_plugin(errored_instances, plugin)
for instance in instances:
    invalid_nodes = plugin.get_invalid(instance)
``` 
I don't know why logging doesn't work in this action so, I added a line at the beginning of each `get_invalid`
`cls.log.debug("This is '%s' and '%s'" % ( cls.__name__ , hou.node(instance.data["instance_node"]) ) )` 

and here's my test case : 
I have 4 pointcaches: 
1) `pointcacheMain` has nor errors 
2) `pointcacheMain2` passed outputsop validation but did pass path validation
3) `pointcacheMain3` didn't pass outputsop (`sop_path` -> None)
4) `pointcacheMain4` didn't pass outputsop (`sop_path` -> `ObjNode`) 

--- 
at the beginning I thought that `Select Invalid` will be called for all plugins 
which is not true! 
however I have a different question now, 
why `Select Invalid` works with all errored instances that are associated with the plugin that action was triggered from 
even if that instance has passed that particular plugin but errored because of another plugin 
> i.e. I think `Select Invalid` on `Invalid output nodes(s)` should have worked only with the listed instances beneath it! 


![image](https://github.com/ynput/OpenPype/assets/20871534/ff41ee5e-6ab2-45c7-bb43-2f75072706f4)

Reply:
Now I'm not sure if that is different issue or it is response to my previous comment?

Reply:
I was writing that comment before but it took me some time to test so that I can have a good visual example of the problem 

> But I agree that there could be some changes that may resolve wrong instances. So it's probably not a good idea.
I agree with that, do you think we should put that example in an issue as a reminder?

Reply:
@BigRoy Checked `pyblish.api` and the plugin from result is the class, so it can be
```
if plugin is not None and result.get("plugin") is not plugin:
```

Reply:
my previous comment was irrelevant sorry,  
I tested it in houdini and it looks good!

Reply:
Thanks @mustafa-zarkash what you described as an issue here with it running over any instance that had **any error** even if not related to the plugin the action is for that is indeed what this PR intends to solve. So if that now works, all good! :)